distributed/client.py

```diff
@@ -3441,8 +3441,11 @@ def upload_file(self, filename, **kwargs):
         """Upload local package to workers
 
         This sends a local file up to all worker nodes.  This file is placed
-        into a temporary directory on Python's system path so any .py,  .egg
-        or .zip  files will be importable.
+        into the working directory of the running worker, see config option
+        `temporary-directory` (defaults to `tempfile.gettempdir`).
```

Comment:
Minor suggestion: Using double backticks with Sphinx and adding `:py:func:` so we link directly to the corresponding Python API docs

```suggestion
        ``temporary-directory`` (defaults to :py:func:`tempfile.gettempdir`).
```